packages/block-editor/src/components/block-selection-clearer/index.js

```diff
@@ -43,7 +39,7 @@ export function useBlockSelectionClearer( onlySelfClicks = false ) {
 				node.removeEventListener( 'mousedown', onMouseDown );
 			};
 		},
-		[ hasSelection, clearSelectedBlock ]
+		[ hasSelectedBlock, hasMultiSelection, clearSelectedBlock ]
```

Comment:
Of course a change in these dependencies won't do anything, but they should remain content anyway. Not sure if we should pass it through `useMergeRefs` after all.